oceannavigator/frontend/src/components/OceanNavigator.jsx

```diff
@@ -178,6 +180,13 @@ function OceanNavigator(props) {
           return !prevCompare;
         });
         break;
+
+      case "enableAnnotationMode":
+        mapRef.current.enableAnnotationMode();
+        break;
+      case "disableAnnotationMode":
+        mapRef.current.disableAnnotationMode();
+        break;
```

Comment:
I think you can combine these into one action since the annotation mode is disabled within the Map component once the user clicks on where they want to place the new label.